daphne/src/testing.rs

```diff
@@ -329,11 +336,8 @@ impl DapAggregator<BearerToken> for MockAggregator {
             .map_err(|e| DapError::Fatal(e.to_string()))?;
         let report_store = report_store_mutex_guard.deref_mut();
 
-        for (inner_bucket_info, store) in report_store.iter_mut() {
-            if task_id == &inner_bucket_info.task_id
-                && batch_interval.start <= inner_bucket_info.window
-                && batch_interval.end() > inner_bucket_info.window
-            {
+        for (id, store) in report_store.iter_mut() {
```

Comment:
This doesn't look right. Why would we mark every report set for the task as collected?

Reply:
The new change fixes this. We are now checking the AggStore.